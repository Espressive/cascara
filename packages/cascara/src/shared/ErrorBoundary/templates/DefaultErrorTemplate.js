import React from 'react';
import PropTypes from 'prop-types';
import { Message } from 'semantic-ui-react';

const DefaultErrorTemplate = ({
  error = new Error(),
  errorInfo = {
    componentStack: '',
  },
  ...rest
}) => (
  <Message
    content={
      <details
        style={{
          fontSize: '12px',
          whiteSpace: 'pre-wrap',
        }}
      >
        {errorInfo.componentStack}
      </details>
    }
    error
    header={error ? error.toString() : 'something went wrong'}
    size='small'
    {...rest}
  />
);
DefaultErrorTemplate.propTypes = {
  error: PropTypes.instanceOf(Error),
  errorInfo: PropTypes.shape({
    componentStack: PropTypes.string.isRequired,
  }),
};

export default DefaultErrorTemplate;
