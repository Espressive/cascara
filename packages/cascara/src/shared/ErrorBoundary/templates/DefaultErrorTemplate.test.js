import React from 'react';
import renderer from 'react-test-renderer';
import DefaultErrorTemplate from './DefaultErrorTemplate';

const props = {
  error: new Error('I crashed!'),
  errorInfo: {
    componentStack: 'Lorem Ipsum is simply dummy text',
  },
};

describe('<DefaultErrorTemplate />', () => {
  test('snapshot for empty state', () => {
    const snapshot = renderer.create(<DefaultErrorTemplate />);
    expect(snapshot).toMatchSnapshot();
  });

  test('snapshot with props', () => {
    const snapshot = renderer.create(<DefaultErrorTemplate {...props} />);
    expect(snapshot).toMatchSnapshot();
  });
});
