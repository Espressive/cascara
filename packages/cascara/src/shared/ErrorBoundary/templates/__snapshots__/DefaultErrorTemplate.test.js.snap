// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DefaultErrorTemplate /> Snapshot for empty state 1`] = `
<div
  className="ui small error message"
>
  <div
    className="content"
  >
    <div
      className="header"
    >
      Error
    </div>
    <details
      style={
        Object {
          "fontSize": "12px",
          "whiteSpace": "pre-wrap",
        }
      }
    >
      
    </details>
  </div>
</div>
`;

exports[`<DefaultErrorTemplate /> Snapshot with props 1`] = `
<div
  className="ui small error message"
>
  <div
    className="content"
  >
    <div
      className="header"
    >
      Error: I crashed!
    </div>
    <details
      style={
        Object {
          "fontSize": "12px",
          "whiteSpace": "pre-wrap",
        }
      }
    >
      Lorem Ipsum is simply dummy text
    </details>
  </div>
</div>
`;
